metabase/metabase
# skip analytics opt out step for people on cloud

# Description

Second part of  [[Epic] Reduce the number of choices in onboarding flow](https://github.com/metabase/metabase/issues/46741).
This PR hides the analytics opt-out for hosted instances

# How to verify

You'll need to copy the "CYPRESS_NO_FEATURES_TOKEN" from 1 password

Remove/backup your app db and run
`MB_PREMIUM_EMBEDDING_TOKEN='the token from above' yarn dev-ee`

You should now be able to do the setup and see that you won't see the analytics opt out step


Note: there will be a broken padding on the last step until  [fix padding on setup completed step when on cloud (+1 −0)](https://github.com/metabase/metabase/pull/47021) is merged

# Tests

Simulating pro/starter on e2e for the setup is not easy, as the `setTokenFeatures` calls an authenticated API that requires admin permissions.
We could do it after the user step, since after that step the user is logged in as admin, but it doesn't seem like a realistic scenario.
We also only have two tokens, the "no feature" ones that actually has the `hosting` feature (to simulate cloud/starter) and the "all features" that has all of them, there would be no way to check for pro self-hosted.
I also didn't want to mock the properties and mock the feature-tokens as it kinda kills the point of testing e2e.

For this reason I decided to go with jest test.
Even though it could have been enough to just update the old jest tests as they were testing the scenario of a pro user (`hosting` and `embedding` feature-tokens), I wanted specifically also test for the starter plan (so only `hosting` feature) and for the "pro but self hosted" scenario to make sure we're not breaking the onboarding.

To simulate all scenarios I decided to go with jest tests and to split the already existing `setup/tests/premium.unit.spec.tsx` in 3 different files:
- `premium-pro-cloud...` both `embedding` and `hosting` features to simulate a pro/ee customer on cloud
- `premium-pro-selfhosted...`: only the `embedding`: feature to simulate a pro/ee customer self-hosting
- `premium-starter-cloud...`: only the `hosting` feature to simulate a starter customer

plus the already existing files:
- `enterprise...`: for ee build but no token
- `common...`: to simulate the oss build

These tests should cover all realistic scenarios, if you notice something is missing please let me know.

In the tests I then check if the analytics opt out step is visible or not, and if the token activation step is visible or not.



This is the recap of the new files and what they test:
<img width="1116" alt="image" src="https://github.com/user-attachments/assets/976bc543-4982-41e0-841c-34df39c9c4fe">
